### PR TITLE
feat(#1): init project

### PR DESCRIPTION
**Pull request by TuranC**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,112 @@
+{
+  "compilerOptions": {
+    //    "incremental": true,
```
Comment (TuranC) [inline tsconfig.json:3]: remove comments

File: package.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    "next": "12.3.1",
+    "prettier": "^2.7.1",
+    "react": "18.2.0",
+    "react-dom": "18.2.0",
+    "storybook": "^6.5.12"
+  },
+  "devDependencies": {
+    "@babel/core": "^7.19.6",
+    "@commitlint/cli": "^17.1.2",
+    "@commitlint/config-conventional": "^17.1.0",
```
Comment (TuranC) [inline package.json:32]: check task number and check message

File: .husky/pre-commit [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+pnpm lint
```
Comment (TuranC) [inline .husky/pre-commit:4]: https://gist.github.com/seeliang/0f0de424d1cdc4541c338f4ee93b7e6a

add git --diff, to optimize lint perfomance

Comment (TuranC) [inline .husky/pre-commit:4]: add to the pre-push hook deepscan and auto tests execution

**Comment by vilnacrm** [comment]
we need to add file .gitignore to all empty directories in the project to ensure that directories are shown in the repository for other developers

**Comment by TuranC** [comment]
fix errors for pnpm install

**Comment by TuranC** [comment]
check MSW, and is it working?

**Comment by TuranC** [comment]
check all booletproof steps

**Review by Kravalg — CHANGES_REQUESTED** [review]
I think that we need to add a docker environment there using the make command like in the [php-service-template.](https://github.com/VilnaCRM-Org/php-service-template/blob/b1def46c86ae89e60a27a526d293a0b53b23c3d4/Makefile)
Then we need to update local and GitHub CI.

**Comment by Kravalg** [comment]
> check MSW, and is it working?

We will introduce this in the next versions

**Pull request merged**